src/deploy_tools/app_builder.py

```diff
@@ -96,3 +102,38 @@ def _create_shell_file(self, app: ShellApp, module: Module) -> None:
             executable=True,
             create_parents=True,
         )
+
+    def _create_binary_file(self, app: BinaryApp, module: Module) -> None:
+        """
+        Download a URL, validate it against its sha256, make it executable
+        and add it to PATH
+        """
+        binary_folder = self._build_layout.get_entrypoints_folder(
+            module.name, module.version
+        )
+        binary_path = binary_folder / app.name
+        binary_path.parent.mkdir(parents=True, exist_ok=True)
+        urlretrieve(app.url, binary_path)
+
+        match app.hash_type:
+            case HashType.SHA256:
+                h = hashlib.sha256()
+            case HashType.SHA512:
+                h = hashlib.sha512()
+            case HashType.MD5:
+                h = hashlib.md5()
+            case HashType.NONE:
+                h = None
+
+        if h is not None:
+            with open(binary_path, "rb") as fh:
+                while True:
+                    data = fh.read(4096)
+                    if len(data) == 0:
+                        break
+                    h.update(data)
+            if h.hexdigest() != app.hash:
+                raise AppBuilderError(f"Downloaded Binary {app.url} digest failure")
+
+        # TODO Martin suggests moving this
+        binary_path.chmod(0o555)
```

Comment:
@MJGaughran Not sure how you want to handle this?

Reply:
Permissions is a magic number here, make a module constant at the very least.

Reply:
Good point. Fixed.